wildmatch.go

```diff
@@ -111,7 +111,7 @@ func slashEscape(p string) string {
 				i += 1
 			}
 		default:
-			pp += string(c)
+			pp += string([]byte{c})
```

Comment:
I think that will end up having the same problem as the original.  `c` is a byte that may be part of a multibyte UTF-8 sequence, but writing `string(c)` interprets it as a rune, causing it to be overencoded.  I expect `EncodeRune` will do the same, since we'll be casting from a byte to a rune.

Ideally, we'd iterate over the string and get a sequence of runes and then build those into a string, but that requires the escaping case to be more complex since we need a state machine to keep track of whether we're in an escape or not.  It also has the problem that it doesn't work for folks that have non-UTF-8 paths, but I'm unsure if that currently works or not.

This would all be much more elegant and less error prone with an iterator and a fold function, but we don't have those.

Reply:
Ah, that makes good sense to me. Thanks for explaining.